components/client/src/main/java/com/hotels/styx/client/netty/connectionpool/NettyToStyxResponsePropagator.java

```diff
@@ -117,23 +117,18 @@ private RuntimeException toStyxException(Throwable cause) {
     }
 
     @Override
-    public void channelInactive(ChannelHandlerContext ctx) throws Exception {
+    public void channelInactive(ChannelHandlerContext ctx) {
         ensureContentProducerIsCreated(ctx);
 
+        TransportLostException cause = new TransportLostException(ctx.channel().remoteAddress(), origin);
+        contentProducer.ifPresent(producer -> producer.channelInactive(cause));
+    }
+
+    private void scheduleResourcesTearDown(ChannelHandlerContext ctx) {
         ctx.channel().eventLoop().schedule(
-                () -> contentProducer.ifPresent(producer -> producer.idleStateEvent(
-                        new ResponseTimeoutException(
-                                origin,
-                                "channelInactive",
-                                producer.receivedBytes(),
-                                producer.receivedChunks(),
-                                producer.emittedBytes(),
-                                producer.emittedChunks()))),
```

Comment:
Please ensure these stats are passed in to the ResponseTimeoutException.
Even better please add a test that verifies that the counters are present in the exception when it is torn down.

Reply:
Or instead log a warning using `FlowControllingHttpContentProducer.warningMessage`. This is the preferred option IMO.

Reply:
The log messages can be tested with `LoggingTestSupport`. Please look for examples of usage in the codebase. It is quite easy to use.

Reply:
Alternatively raise a separate GitHub enhancement issue.